src/main/java/com/ipiecoles/java/java350/service/EmployeService.java

```diff
@@ -104,33 +165,24 @@ public void calculPerformanceCommercial(String matricule, Long caTraite, Long ob
             throw new EmployeException("Le matricule " + matricule + " n'existe pas !");
         }
 
-        Integer performance = Entreprise.PERFORMANCE_BASE;
-        //Cas 2
-        if(caTraite >= objectifCa*0.8 && caTraite < objectifCa*0.95){
-            performance = Math.max(Entreprise.PERFORMANCE_BASE, employe.getPerformance() - 2);
-        }
-        //Cas 3
-        else if(caTraite >= objectifCa*0.95 && caTraite <= objectifCa*1.05){
-            performance = Math.max(Entreprise.PERFORMANCE_BASE, employe.getPerformance());
-        }
-        //Cas 4
-        else if(caTraite <= objectifCa*1.2 && caTraite > objectifCa*1.05){
-            performance = employe.getPerformance() + 1;
-        }
-        //Cas 5
-        else if(caTraite > objectifCa*1.2){
-            performance = employe.getPerformance() + 4;
-        }
-        //Si autre cas, on reste à la performance de base.
+        ArrayList<PerformanceRule> rules = new ArrayList<>();
+        rules.add(new PerformanceRule(null, objectifCa * 0.8, -employe.getPerformance()));
+        rules.add(new PerformanceRule(objectifCa * 0.8, objectifCa * 0.95, -2));
+        rules.add(new PerformanceRule(objectifCa * 0.95, objectifCa * 1.05 + 1, 0));
+        rules.add(new PerformanceRule(objectifCa * 1.05 + 1,  objectifCa * 1.2 + 1, 1));
+        rules.add(new PerformanceRule(objectifCa * 1.2 + 1, null, 4));
+
+        rules.forEach(rule ->{ if (rule.verify(caTraite)) {
```

Comment:
Eventuellement utiliser le filter au lieu du if dans ton forEach 
```suggestion
        rules.filter(rule -> rule.verify(caTraite)).forEach(rule ->{{
```